src/components/widgets/AutocompleteWidget/AutocompleteWidget.tsx

```diff
@@ -121,83 +125,126 @@ function AutocompleteWidget(props: AutocompleteWidgetProps) {
+                                            iri: selection.iri,
+                                            label: selection.label,
+                                            ontology_name: selection.ontology_name,
+                                            type: selection.type,
+                                            short_form: selection.short_form,
+                                        },
                                     },
-                                },
-                            ]);
-                            setLoadingState(false);
-                        }
-                    })
-                }
-            });
-        } else if (props.selectOption?.label && props.allowCustomTerms) { // when a custom term is passed
-            setLoadingState(true);
-            setOptions([
-                {
-                    label: props.selectOption?.label,
-                    value: {
-                        iri: "",
-                        label: "",
-                        ontology_name: "",
-                        type: "",
-                        short_form: "",
-                    }
-                },
-            ]);
-            setSelectedOptions([
-                {
-                    label: props.selectOption?.label,
-                    value: {
-                        iri: "",
-                        label: "",
-                        ontology_name: "",
-                        type: "",
-                        short_form: "",
+                                ]);
+                            }
+                        })
                     }
-                },
-            ]);
-            setLoadingState(false);
+                });
+            } else if (props.selectOption?.label && props.allowCustomTerms) { // when a custom term is passed
+                setOptions([
+                    {
+                        label: props.selectOption?.label,
+                        value: {
+                            iri: "",
+                            label: "",
+                            ontology_name: "",
+                            type: "",
+                            short_form: "",
+                        }
+                    },
+                ]);
+                setSelectedOptions([
+                    {
+                        label: props.selectOption?.label,
+                        value: {
+                            iri: "",
+                            label: "",
+                            ontology_name: "",
+                            type: "",
+                            short_form: "",
+                        }
+                    },
+                ]);
+            }
         }
-    }, []); // no dependencies - does only need to be executed once when mounting the component @TODO: Dependency on props.selectOption
+    )
 
+    /**
+     * fetches new options when searchValue changes (replaces method onSearchChange)
```

Comment:
I would remove the part in brackets, because it refers to a method which doesn't exist anymore

Reply:
Done